test/lib/mayaUsd/fileio/testAddMayaReference.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2022 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import fixturesUtils
+
+import mayaUsd
+import mayaUtils
+
+from maya import cmds
+from maya import standalone
+from mayaUsdLibRegisterStrings import getMayaUsdLibString
+from mayaUSDRegisterStrings import getMayaUsdString
+
+import unittest
+
+class AddMayaReferenceTestCase(unittest.TestCase):
+    '''Test Add Maya Reference.
+    '''
+    pluginsLoaded = False
+
+    @classmethod
+    def setUpClass(cls):
+        fixturesUtils.readOnlySetUpClass(__file__, loadPlugin=False)
+        if not cls.pluginsLoaded:
+            cls.pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
+
+    @classmethod
+    def tearDownClass(cls):
+        standalone.uninitialize()
+
+    def setUp(self):
+        cmds.file(new=True, force=True)
+
+    def createSimpleMayaScene(self):
+        import os
+        import maya.cmds as cmds
+        import tempfile
+
+        cmds.CreatePolygonSphere()
+        tempMayaFile = os.path.join(tempfile.gettempdir(), 'simpleSphere.ma')
+        cmds.file(rename=tempMayaFile)
+        cmds.file(save=True, force=True, type='mayaAscii')
+        return tempMayaFile
+
+    def testDefineInVariant(self):
+        '''Test the "Define in Variant" options for Add Maya Reference.'''
+
+        # First we need a pure Maya scene to reference in.
+        mayaSceneStr = self.createSimpleMayaScene()
```

Comment:
Shouldn't we clean this up once the test is done?

Reply:
No need really. It is created in a folder (specific to this test) in the "temporary" testing folder for all tests.